pages/api/patients/id.ts

```diff
@@ -0,0 +1,26 @@
+// Next.js API route support: https://nextjs.org/docs/api-routes/introduction
+import type { NextApiRequest, NextApiResponse } from 'next'
+import prisma from '../../../lib/prisma'
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  if (req.method === 'POST') {
+    try {
+      let { id } = req.body.data
+
+      var patients = await prisma.patient.findUniqueOrThrow({
+        where: {
+          id: id
+        }
+      })
```

Comment:
this should be immutable in the local context. also if we are looking for a unique it should be `patient`. This will also throw an error I believe so we need to wrap it in a try-catch so we don't have a server error.

```suggestion
      let patient = null
      try {
      patient = await prisma.patient.findUniqueOrThrow({
        where: {
          id: id
        }
      })
      }
      catch(e){
      res.status(404).json({ message: "User not found", error: e })
      }
```

Reply:
I think we shouldn't nest too many try-catches. Currently if this does throw, it'll throw a notFoundError which will be caught by the existing catch block, and the message will reflect the reason for error so I think it should be fine